### PR TITLE
[common/streams] use arrow2 bitmap to improve memory usage

### PR DESCRIPTION
**Pull request by cppcoffee**
I hereby agree to the terms of the CLA available at: https://datafuse.rs/policies/cla/

## Summary

Use arrow2 bitmap to improve memory usage

## Changelog

- Improvement

## Related Issues

None

## Test Plan

Unit Tests

Stateless Tests

### CODE REVIEW

File: common/streams/src/stream_limit_by.rs [hunk 1]
```diff
@@ -50,20 +52,19 @@ impl LimitByStream {
     }
 
     pub fn limit_by(&mut self, block: &DataBlock) -> Result<Option<DataBlock>> {
-        // TODO: use BitVec here.
```
Comment (jorgecarleitao) [inline common/streams/src/stream_limit_by.rs:53]: I would suggest to use `arrow2::bitmap::MutableBitmap` here and then use `BooleanArray::from_data(filter_vec.into(), None)`. This should be at least 10x faster, as we avoid running over all bits twice.

```rust
let mut filter_vec = arrow2::bitmap::MutableBitmap::from_zero_len(block.num_rows())
...
filter_vec.set(row as usize, *count <= self.limit)
...

let array = arrow2::array::BooleanArray::from_data(filter_vec.into(), None);
// convert to DFBooleanArray
```

`bitvec` and `arrow2::bitmap` (`MutableBitmap` specifically) overlap in functionality, but creating a bitmap directly from arrow avoids a full scan over all bits and an extra memory allocation. :)

Comment (cppcoffee) [reply common/streams/src/stream_limit_by.rs:53]: Ok, should I open a new PR or modify it in the current one?

Comment (BohuTANG) [reply common/streams/src/stream_limit_by.rs:53]: Modify is the simple way?

Comment (cppcoffee) [reply common/streams/src/stream_limit_by.rs:53]: Ok, I'll fix it.

**Review by BohuTANG — APPROVED** [review]
LGTM
Thank you @cppcoffee

**Pull request merged**